scripts/emerge/ingest_to_tdr.py

```diff
@@ -21,84 +21,116 @@ def get_access_token():
-    retrieveJob_status_code = retrieveJob_response.status_code
 
-    # job_status = response_json['job_status']
-    if retrieveJob_status_code == 202:                                           # still running
-        print(f"load job {job_id} --> running")
-        return retrieveJob_status_code, retrieveJob_response_json
+    response = requests.get(uri, headers=headers)
+    response_json = json.loads(response.text)
+    status_code = response.status_code
 
-    if retrieveJob_status_code == 200:                                          # finished successfully
-        print(f"load job {job_id} --> success")
-        retrieveJobResult_response = requests.get(uri + "/result", headers=headers)
-        retrieveJobResult_response_json = json.loads(retrieveJobResult_response.text)
-        retrieveJobResult_status_code = retrieveJobResult_response.status_code
-        return retrieveJobResult_status_code, retrieveJobResult_response_json
+    return status_code, response_json
 
-    print(f"load job {job_id} --> not successful")  # finished but not successfully
-    return retrieveJob_status_code, retrieveJob_response_json
 
+def get_job_status(job_id):
+    """retrieveJobStatus"""
 
-def load_data(dataset_id, ingest_data):
-    """Load data into TDR"""
+    # first check job status - retrieveJob
+    uri = f"https://data.terra.bio/api/repository/v1/jobs/{job_id}"
+    
+    headers = {"Authorization": "Bearer " + get_access_token(),
+               "accept": "application/json"}
+
+    response = requests.get(uri, headers=headers)
+    response_json = json.loads(response.text)
+    status_code = response.status_code
+
+    return status_code, response_json
+
+
+def ingest_dataset(dataset_id, data):
+    """Load data into TDR with ingestDataset."""
 
     uri = f"https://data.terra.bio/api/repository/v1/datasets/{dataset_id}/ingest"
 
     headers = {"Authorization": "Bearer " + get_access_token(),
                "accept": "application/json",
                "Content-Type": "application/json"}
 
-    response = requests.post(uri, headers=headers, data=ingest_data)
+    response = requests.post(uri, headers=headers, data=data)
     status_code = response.status_code
 
-    if status_code != 202:
-        return response.text
+    if status_code != 202:  # if ingest start-up fails
+        raise ValueError(response.text)
 
+    # if ingest start-up succeeds
     return json.loads(response.text)
 
 
-def call_ingest_dataset(control_file_path, target_table_name, dataset_id):
+def create_ingest_dataset_request(control_file_path, target_table_name, load_tag=None):
+    """Create the ingestDataset request body."""
+
+    load_dict = {"format": "json",
+                 "path": control_file_path,
+                 "table": target_table_name,
+                 "resolve_existing_files": "true",
+                 "updateStrategy": "replace"
+                }
+    # if user provides a load_tag, add it to request body
```

Comment:
what does including a load tag do?

Reply:
cool, thanks! i think our future selves would benefit from a comment somewhere to explain this briefly, maybe in the argparse section, like "allows ingest of duplicate files in separate ingest calls when using the same load tag"